client/client.go

```diff
@@ -133,19 +131,88 @@ func (client *Client) AppendToStream(
 	}, nil
 }
 
+func (client *Client) SetStreamMetadata(
+	context context.Context,
+	streamID string,
+	opts *options.AppendToStreamOptions,
+	metadata esdb_metadata.StreamMetadata,
+) (*WriteResult, error) {
+	streamName := fmt.Sprintf("$$%v", streamID)
+	props, err := metadata.ToMap()
+
+	if err != nil {
+		return nil, err
+	}
+
+	event, err := messages.NewJsonProposedEvent("$metadata", props)
+
+	if err != nil {
+		return nil, err
+	}
+
+	result, err := client.AppendToStream(context, streamName, opts, event)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (client *Client) GetStreamMetadata(
+	context context.Context,
+	streamID string,
+	opts *options.ReadStreamEventsOptions,
+) (*esdb_metadata.StreamMetadata, error) {
+	streamName := fmt.Sprintf("$$%v", streamID)
```

Comment:
why "$$%v"? `"$$%s"` should be fine as streamID is string (or even `"$$" + streamID`)

Reply:
That's an internal name convention to point at the stream metadata stream.